mem0/vector_stores/valkey.py

```diff
@@ -0,0 +1,889 @@
+            vector_size (int, optional): Size of the vector embeddings. Defaults to None, which uses the current embedding_model_dims.
+            distance (str, optional): Distance metric to use. Defaults to None, which uses 'cosine'.
+
+        Returns:
+            The created index object.
+        """
+        # Use provided parameters or fall back to instance attributes
+        collection_name = name or self.collection_name
+        embedding_dims = vector_size or self.embedding_model_dims
+        distance_metric = distance or "COSINE"
+        prefix = f"mem0:{collection_name}"
+
+        # Try to drop the index if it exists
+        try:
+            self._drop_index(collection_name)
+        except Exception as e:
+            logger.warning(f"Error dropping existing index {collection_name}: {e}")
+
+        # Build the vector field configuration based on index type
+        if self.index_type == "hnsw":
+            vector_config = [
+                "embedding",
+                "VECTOR",
+                "HNSW",
+                "12",  # Attribute count: TYPE, FLOAT32, DIM, dims, DISTANCE_METRIC, metric, M, m, EF_CONSTRUCTION, ef_construction, EF_RUNTIME, ef_runtime
+                "TYPE",
+                "FLOAT32",
+                "DIM",
+                str(embedding_dims),
+                "DISTANCE_METRIC",
+                distance_metric,
+                "M",
+                str(self.hnsw_m),
+                "EF_CONSTRUCTION",
+                str(self.hnsw_ef_construction),
+                "EF_RUNTIME",
+                str(self.hnsw_ef_runtime),
+            ]
+        else:  # flat
+            vector_config = [
+                "embedding",
+                "VECTOR",
+                "FLAT",
+                "6",  # Attribute count: TYPE, FLOAT32, DIM, dims, DISTANCE_METRIC, metric
+                "TYPE",
+                "FLOAT32",
+                "DIM",
+                str(embedding_dims),
+                "DISTANCE_METRIC",
+                distance_metric,
+            ]
+
+        # Create the index using raw command
+        cmd = [
+            "FT.CREATE",
+            collection_name,
+            "ON",
+            "HASH",
+            "PREFIX",
+            "1",
+            prefix,
+            "SCHEMA",
+            "memory_id",
+            "TAG",
+            "SEPARATOR",
+            ",",
+            "hash",
+            "TAG",
+            "SEPARATOR",
+            ",",
+            "agent_id",
+            "TAG",
+            "SEPARATOR",
+            ",",
+            "run_id",
+            "TAG",
+            "SEPARATOR",
+            ",",
+            "user_id",
+            "TAG",
+            "SEPARATOR",
+            ",",
+            "memory",
+            "TAG",
+            "SEPARATOR",
+            ",",
+            "metadata",
+            "TAG",
+            "SEPARATOR",
+            ",",
```

Comment:
```suggestion
            "hash",
            "TAG",
            "agent_id",
            "TAG",
            "run_id",
            "TAG",
            "user_id",
            "TAG",
            "memory",
            "TAG",
            "metadata",
            "TAG",
```
The separator is the default, you can omit it.

Reply:
Done. I have also refactored to remove duplication in index creation